protocol/src/main/java/io/camunda/zeebe/protocol/record/value/TenantOwned.java

```diff
@@ -18,6 +18,12 @@
 /** Concerns an entity that is owned by a tenant. */
 public interface TenantOwned {
 
+  /**
+   * The default tenant identifier. When multi-tenancy is disabled, entities are owned by this
+   * tenant. This is done in case multi-tenancy is enabled in the future.
+   */
+  String DEFAULT_TENANT_IDENTIFIER = "<default>";
```

Comment:
Setting `NIL_VALUE` as the default tenant identifier might be problematic, as all data (existing or new) must be assigned to a tenant (and this is a hard requirement). Fruthermote, Operate and Optimize already implemented data migrations with the `<default>` tenant, so we would set them back if we ask them for further changes.

I would suggest adding a note about this to https://github.com/camunda/zeebe/issues/13315 for now. Since this is a data migration issue we should tackle it then.

One solution that pops in my mind is "lazy migration". We use `NIL_VALUE` as the default tenant identifier while multi-tenancy is disabled. Once multi-tenancy is enabled, the first read from the state of data that has `NIL_VALUE` for the `tenantId` triggers a replacement (either directly in the state, or we just provide the `<default>` tenant id instead).

WDYT?

Reply:
I noted about this discussion in #13315.